app/controllers/application_controller/ci_processing.rb

```diff
@@ -1868,6 +1868,7 @@ def process_cloud_object_storage_buttons(pressed)
     task = pressed.sub("#{klass.name.underscore.to_sym}_", "")
 
     return tag(klass) if task == "tag"
+    return javascript_redirect(:controller => klass.name.underscore.to_sym, :action => "new") if task == "new"
 
```

Comment:
This does not seem right.
This logic needs to be in the `cloud_object_store_container` controller, not here.

Can you look up a few examples and fix this?